pkg/tuple/tuple.go

```diff
@@ -33,7 +33,7 @@ const (
 var (
 	userIDRegex   = regexp.MustCompile(`^[^:#\s]+$`)
 	objectRegex   = regexp.MustCompile(`^[^:#\s]+:[^#:\s]+$`)
-	userSetRegex  = regexp.MustCompile(`^[^:#\s]+:[^#\s]+#[^:#\s]+$`)
+	userSetRegex  = regexp.MustCompile(`^[^:#\s]+:[^#:*\s]+#[^:#*\s]+$`)
```

Comment:
Similar to https://github.com/openfga/language/blob/main/pkg/go/validation/validation-rules.go#L32

```go
"^[^:#@\\*\\s]{1,254}:[^#:\\*\\s]+#[^:#@\\*\\s]{1,50}$",
```

Reply:
do we consider `*:|#|` a valid userset? It certainly fits the pattern in this line of code.

my observation may be beyond the scope of this PR, but I understand that there are some implied constraints that aren't represented properly in our regex patterns.

are we just doing "partial" validation with this pattern? is the evaluation of this pattern dependent on other validations having been evaluated before or after this pattern is evaluated?

Reply:
ah, we are just matching here to determine what it is...

Reply:
> do we consider *:|#| a valid userset?

no, because `*` is not a valid type 😅 i'll have to amend this

> we are just matching here to determine what it is... I feel like that is different than a validation check.

this regex is validating if a string is a valid userset, it's not enough to say whether it's a valid _user_, and it's not enough to say whether it's a valid userset _for the current model_

Reply:
If that is true, the regex is not checking for character length or many characters that are likely invalid such as Unicode control characters like U+0000 (null). I wouldn't trust that a value is entirely valid after being matched against this pattern.

Reply:
Ah, this code assumes prior validation: https://github.com/openfga/api/blob/c96ec773cae9caf2a9536856bf1fa206a42ef7ef/openfga/v1/openfga_service.proto#L1134-L1141

Reply:
@senojj opened https://github.com/openfga/openfga/issues/1876 to track that work.